elizaOS/eliza
# fix: disable trust provider for PostGres db

# Relates to:

- Issues reported by using of crashes due to this error `Error in trust score provider: Cannot read properties of undefined (reading 'prepare')`

# Risks

- low risk

# Background

- Many users in discord have reported this error when using the Solana plugin and PostGres db. The trust provider tries to run but fails because the `TrustScoreDatabase` only works with Sqllite.

## What does this PR do?

- Prevents the error from occurring by adding a check for Postgres and logging a warning instead of crashing

## What kind of change is this?

- bug fix





## Why are we doing this? Any context or related work?

- Allow users of PostGres to utilize Solana and Starknet plugin

# Documentation changes needed?

- No





# Testing

- Set up a PostGres Connection
- Activate the Solana plugin 
- Ask the agent questions about recommendations such as 'Should I buy SOL?'

BEFORE

`Error in trust score provider: Cannot read properties of undefined (reading 'prepare')`

AFTER

`skipping trust evaluator because db is postgres`
